src/map/chrif.cpp

```diff
@@ -1649,12 +1649,13 @@ int chrif_bsdata_save(struct map_session_data *sd, bool quit) {
 
 	// Removing...
 	if (quit && sd->bonus_script.head) {
-		uint16 flag = BSF_REM_ON_LOGOUT; //Remove bonus when logout
+		uint32 flag = BSF_REM_ON_LOGOUT; //Remove bonus when logout
+
 		if (battle_config.debuff_on_logout&1) //Remove negative buffs
 			flag |= BSF_REM_DEBUFF;
 		if (battle_config.debuff_on_logout&2) //Remove positive buffs
 			flag |= BSF_REM_BUFF;
-		pc_bonus_script_clear(sd, flag);
+		pc_bonus_script_clear(sd, static_cast<e_bonus_script_flags>(flag));
```

Comment:
Should not be casted here, since it is used as a bitmask and not enumerated values might be inside.